commit 3026c19a14ba71ccd4dc4925abab9395ea12839c
Author: Steve Wise <swise@opengridcomputing.com>
Date:   Wed May 27 14:43:39 2009 -0700

    RDMA/cxgb3: Limit fast register size based on T3 limitations
    
    T3 firmware only supports one WRs worth of page list for fast register
    work requests.  The driver currently allows 2 WRs worth, which
    doesn't work for T3, so reduce the limit in the driver.
    
    Signed-off-by: Steve Wise <swise@opengridcomputing.com>
    Signed-off-by: Roland Dreier <rolandd@cisco.com>

diff --git a/drivers/infiniband/hw/cxgb3/cxio_wr.h b/drivers/infiniband/hw/cxgb3/cxio_wr.h
index ff9be1a..32e3b14 100644
--- a/drivers/infiniband/hw/cxgb3/cxio_wr.h
+++ b/drivers/infiniband/hw/cxgb3/cxio_wr.h
@@ -176,7 +176,7 @@ struct t3_send_wr {
 	struct t3_sge sgl[T3_MAX_SGE];	/* 4+ */
 };
 
-#define T3_MAX_FASTREG_DEPTH 24
+#define T3_MAX_FASTREG_DEPTH 10
 #define T3_MAX_FASTREG_FRAG 10
 
 struct t3_fastreg_wr {
